File: client/src/main/java/com/flipkart/grayskull/hooks/NoOpRefreshHandlerRef.java
package com.flipkart.grayskull.hooks;

import lombok.extern.slf4j.Slf4j;

/**
 * Placeholder implementation of RefreshHandlerRef.
 * This allows clients to register hooks without errors, but the hooks are not invoked
 * until full server sent events support is implemented in a future version.
 */
@Slf4j
public final class NoOpRefreshHandlerRef implements RefreshHandlerRef {
    
    /**
     * Singleton instance of the no-op refresh hook handle.
     */
    public static final NoOpRefreshHandlerRef INSTANCE = new NoOpRefreshHandlerRef();
    
    private NoOpRefreshHandlerRef() {
        // Private constructor to prevent instantiation
    }

    @Override
    public String getSecretRef() {
        return "";
    }

    @Override
    public boolean isActive() {
        return false;
    }

    @Override
    public void unRegister() {
        log.debug("Unregister called on no-op refresh hook handle (placeholder implementation)");
        // No-op
    }
}

File: clients/go/client-impl/hooks/noOpRefreshHandlerRef.go
package hooks

import "log"

// NoOpRefreshHandlerRef is a placeholder implementation of RefreshHandlerRef.
// This allows clients to register hooks without errors, but the hooks are not invoked
// until full server sent events support is implemented in a future version.
type NoOpRefreshHandlerRef struct{}

var (
	// Instance is the singleton instance of the no-op refresh hook handle.
	Instance = &NoOpRefreshHandlerRef{}
)

// GetSecretRef returns an empty string as this is a no-op implementation.
func (n *NoOpRefreshHandlerRef) GetSecretRef() string {
	return ""
}

// IsActive always returns false as this is a no-op implementation.
func (n *NoOpRefreshHandlerRef) IsActive() bool {
	return false
}

// Unregister logs a debug message as this is a no-op implementation.
func (n *NoOpRefreshHandlerRef) Unregister() {
	log.Println("Unregister called on no-op refresh hook handle (placeholder implementation)")
	// No-op
}
